api/lifetime_watcher.go

```diff
@@ -270,6 +270,8 @@ func (r *LifetimeWatcher) doRenew() error {
 
 		var leaseDuration time.Duration
```

Comment:
This is inherited, but there are a lot of duration variables, and I think renaming this one to `remainingLeaseDuration` would help make its intention and mutability a bit clearer. If you think that's too long, `lease` and `duration` are kind of implying the same concept each, and perhaps we could drop one of those words from each of the variables. WDYT?

Reply:
I didn't like the suggestion when I read it, but I tried making the change and I liked it.